packages/space-opera/src/components/camera_settings/camera_settings.ts

```diff
@@ -153,79 +131,57 @@ export class CameraSettings extends ConnectedLitElement {
   static styles = cameraSettingsStyles;
 
   @internalProperty() config: ModelViewerConfig = {};
-  @internalProperty() camera: Camera = INITIAL_CAMERA;
-  @internalProperty() initialCamera: Camera = INITIAL_CAMERA;
-  @internalProperty() cameraOutOfBounds: boolean = false;
 
-  @query('me-camera-orbit-editor') cameraOrbitEditor?: CameraOrbitEditor;
+  @query('me-camera-orbit-editor') cameraOrbitEditor!: CameraOrbitEditor;
+  @query('me-camera-target-input') cameraTargetInput!: CameraTargetInput;
   @query('me-checkbox#auto-rotate') autoRotateCheckbox!: CheckboxElement;
 
   // Specifically overriding a super class method.
   // tslint:disable-next-line:enforce-name-casing
   async _getUpdateComplete() {
     await super._getUpdateComplete();
-    await this.cameraOrbitEditor!.updateComplete;
+    await this.cameraOrbitEditor.updateComplete;
     await this.autoRotateCheckbox.updateComplete;
   }
 
   stateChanged(state: State) {
-    this.config = getConfig(state);
-    this.camera = getCamera(state);
-    this.cameraOutOfBounds = this.outOfBounds();
-  }
-
-  orbitValueBound(limits: Limits|undefined, val: string|number) {
-    if ((limits !== undefined) &&
-        ((limits.max !== 'auto' && val > limits.max) ||
-         (limits.min !== 'auto' && val < limits.min))) {
-      return true;
+    const config = getConfig(state);
+    if (config !== this.config) {
+      this.config = config;
+      this.updateInitialCamera();
     }
-    return false;
   }
 
-  outOfBounds() {
-    const snippet = this.camera;
-    if (snippet.orbit === undefined) {
-      return false;
-    }
-    if (this.orbitValueBound(snippet.pitchLimitsDeg, snippet.orbit?.phiDeg)) {
-      return true;
-    } else if (this.orbitValueBound(
-                   snippet.yawLimitsDeg, snippet.orbit?.thetaDeg)) {
-      return true;
-    } else if (this.orbitValueBound(
-                   snippet.radiusLimits, snippet.orbit?.radius)) {
-      return true;
+  async updateInitialCamera() {
+    const cameraState = await getCameraState();
+    this.cameraTargetInput.target = cameraState.target;
+    if (this.config.cameraOrbit == null) {
+      this.cameraOrbitEditor.style.display = 'none';
+    } else {
+      const currentOrbit = cameraState.orbit;
+      this.cameraOrbitEditor.yawInput.value = currentOrbit.thetaDeg;
+      this.cameraOrbitEditor.pitchInput.value = currentOrbit.phiDeg;
+      this.cameraOrbitEditor.style.display = '';
```

Comment:
Question: what is the difference of having display set to none vs ''?

Reply:
setting an attribute to `''` mean unset, so it now just inherits its behavior from its parent HTML elements. I could set it to `'block'`, but this  is safer, since I don't have to worry about what it's originally computed style  was. In this context, `''` means show.